src/material-experimental/mdc-button-toggle/testing/button-toggle-harness.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ButtonToggleHarnessFilters} from './button-toggle-harness-filters';
+// import {MatButtonToggleAppearance} from '@angular/material/button-toggle';
+
+
+/** Harness for interacting with a MDC-based mat-button-toggle in tests. */
+export class MatButtonToggleHarness extends ComponentHarness {
+  /** The selector for the host element of a `MatButtonToggle` instance. */
+  static hostSelector = '.mat-mdc-button-toggle';
+
+  private _label = this.locatorFor('.mdc-button-toggle__label');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatButtonToggleHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which button toggle instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ButtonToggleHarnessFilters = {}): HarnessPredicate<MatButtonToggleHarness> {
+    return new HarnessPredicate(MatButtonToggleHarness, options)
+        .addOption('text', options.text,
+            (harness, text) => HarnessPredicate.stringMatches(harness.getText(), text))
+        .addOption('name', options.name,
+            (harness, name) => HarnessPredicate.stringMatches(harness.getName(), name))
+        .addOption('checked', options.checked,
+            async (harness, checked) => (await harness.isChecked()) === checked);
+  }
+
+  /** Gets a boolean promise indicating if the button toggle is checked. */
+  async isChecked(): Promise<boolean> {
+    const checked = (await this.host()).getAttribute('aria-pressed');
+    return coerceBooleanProperty(await checked);
+  }
+
+  /** Gets a boolean promise indicating if the button toggle is disabled. */
+  async isDisabled(): Promise<boolean> {
+    const disabled = (await this.host()).getAttribute('disabled');
+    return coerceBooleanProperty(await disabled);
+  }
+
+  /** Gets a promise for the button toggle's name. */
+  async getName(): Promise<string | null> {
+    return (await this.host()).getAttribute('name');
+  }
+
+  /** Gets a promise for the button toggle's aria-label. */
+  async getAriaLabel(): Promise<string | null> {
+    return (await this.host()).getAttribute('aria-label');
+  }
+
+  /** Gets a promise for the button toggles's aria-labelledby. */
+  async getAriaLabelledby(): Promise<string | null> {
+    return (await this.host()).getAttribute('aria-labelledby');
+  }
+
+  // /** Gets the appearance that the button toggle is using. */
```

Comment:
you can either: 
- uncomment this, mark it deprecated, and `return 'standard'`
- just delete it